### ISSUE TITLE
Submit button javascript error on submission of getting started survey.

### ISSUE THREAD

**Issue by pmanko** [myapnea/www.myapnea.org#655]
This error pops up on certain user flows through the survey. I had it happen on both Firefox and Chrome when creating a new user and going through the about-me getting started survey.

Refreshing the page fixes the problem - the submit button works again.

Related screen shots:
![screenshot from 2015-05-13 13 18 30](https://cloud.githubusercontent.com/assets/711826/7616628/df4f65ea-f973-11e4-83ab-aab14cf489af.png)
![screenshot from 2015-05-13 13 17 30](https://cloud.githubusercontent.com/assets/711826/7616629/df4fb7ac-f973-11e4-9b8c-09494833115b.png)

**Comment by pmanko** [comment]
Also, I cannot replicate it outside the `getting-started` user flow. All the other survey submit buttons seem to work after finishing the last question.

**Comment by pmanko** [comment]
Another reason why this bug might have been difficult to replicate before:

After successfully pressing submit buttons on other surveys, the error could not be replicated until I cleared the browser cache and saved files.

**Comment by pmanko** [comment]
checkCompletion() is returning false when this happens, even though all the questions are completed

**Comment by pmanko** [comment]
'$("[data-object~='date--error-message']").length > 0 && $("[data-object~='date--error-message']").html() !== ""'

returns true, even though the date field does not have any errors anymore

**Comment by pmanko** [comment]
@krand7 @remomueller The problem seemed to be with the date widget errors. If an error in the date format was returned, it would be displayed under the widget in red text. When the error was fixed, the container for the error would be hidden, but the error text would still be present. 

The `checkCompletion()` method checked if the text was present, but not whether the error was hidden. 

I added this second check. However, @krand7  the actual error message shown `Cannot read property top of undefined` in the method `nextQuestionScroll` should probably be more gracefully handled in case any other series of events causes this error to occur.
